site/components/src/main/java/uk/nhs/hee/web/components/MiniHubComponent.java

```diff
@@ -1,61 +1,69 @@
 package uk.nhs.hee.web.components;
 
+import org.hippoecm.hst.content.beans.standard.HippoBean;
 import org.hippoecm.hst.core.component.HstRequest;
 import org.hippoecm.hst.core.component.HstResponse;
 import org.hippoecm.hst.core.parameters.ParametersInfo;
 import org.onehippo.cms7.essentials.components.EssentialsDocumentComponent;
-import uk.nhs.hee.web.beans.Guidance;
 import uk.nhs.hee.web.beans.MiniHub;
+import uk.nhs.hee.web.beans.MinihubSection;
 import uk.nhs.hee.web.components.info.MiniHubComponentInfo;
 
 import java.util.List;
 
+/**
+ * Base component for the MiniHub Page.
+ */
 @ParametersInfo(type = MiniHubComponentInfo.class)
 public class MiniHubComponent extends EssentialsDocumentComponent {
+
     @Override
     public void doBeforeRender(final HstRequest request, final HstResponse response) {
         super.doBeforeRender(request, response);
```

Comment:
Two observations 
1.  When request is null or empty, the setmodel function will raise an error. So you can add ( if request !=null ), then your action
2. general error handling exception is missing, try to catch all exceptions . For example, on this statement document.getParentBean().getChildBeans(MiniHub.class).get(0), it goes three levels to retrieve the document.

Reply:
> 1. When request is null or empty, the setmodel function will raise an error. So you can add ( if request !=null ), then your action

1. I've followed the same convention Selva/Manifesto have used for the other's. Under what situation do you expect for the actual request object to be null, and what should be the behaviour when such instances are encountered? @adaMiclos any views?

Reply:
> 2\. general error handling exception is missing, try to catch all exceptions . For example, on this statement document.getParentBean().getChildBeans(MiniHub.class).get(0), it goes three levels to retrieve the document.

Good catch, may need a bit of help from you and @adaMiclos to capture the relevant exceptions and what to do when such exceptions are caught, as am not sure if catching just Exception is good practice.. @adaMiclos any thoughts?

Reply:
1. Usually frameworks take care of the request object so that is never null. I don't think we need to check for that.
2. Yeah, it might be good to add null checks for the parentBean and length check for the childBeans list. We could log an error message and set the document model to null or Optional.

Reply:
Thanks both, I've just pushed commit [2787488](https://github.com/Manifesto-Digital/hee-cms-platform/pull/180/commits/27874883c5218a6af369a532d6f64db0a1a93abd). Can you take a look please.